gcp/volume.go

```diff
@@ -141,8 +142,48 @@ type mountPoints struct {
 }
 
 func (c *Client) getVolumeByID(volume volumeRequest) (volumeResult, error) {
+	var baseURL string
+	var originalID string = ""
+
+	// terraform import will specify volumeID.
+	// Issue is, that volumeID is unqiue per region, but might exist in different regions in same project.
+	// For that case, we will support a special ID format for terraform import ADDR ID.
+	// ID = <volumeID>:<region>
+	s := strings.Split(volume.VolumeID, ":")
+	if len(s) == 2 {
+		originalID = volume.VolumeID
+		volume.VolumeID = s[0]
+		volume.Region = s[1]
+	}
+
+	if volume.Region == "" {
+		// terraform import: ID = <volumeID> and no region specified
+		// find all volumes which match VolumeID
+		volumes, err := c.filterAllVolumes(func(v volumeResult) bool {
+			return v.VolumeID == volume.VolumeID
+		})
+		if err != nil {
+			return volumeResult{}, err
+		}
 
-	baseURL := fmt.Sprintf("%s/Volumes/%s", volume.Region, volume.VolumeID)
+		if len(volumes) == 1 {
+			// we found the right volume to import
+			volume.Region = volumes[0].Region
+		} else {
+			if len(volumes) == 0 {
+				return volumeResult{}, fmt.Errorf("getVolumeByID: No volume found with ID %s", volume.VolumeID)
+			}
+			if len(volumes) > 1 {
+				// return error message which tells user to rerun with ID = <volumeID>:<region> format
+				return volumeResult{}, fmt.Errorf(`getVolumeByID: More than one volume found with ID %s. \n
+				If this happend while running terraform import, please use \n
+				terraform import ADDR ID, with ID using <volumeID>:<region_name> format`, volume.VolumeID)
+			}
+			return volumeResult{}, fmt.Errorf("getVolumeByID: WTF error with volume ID %s", volume.VolumeID)
```

Comment:
WTF?

WTF -> Unexpected 


Though I'm not sure it makes sense anyway.   len is  >= 0,  so we exhausted all cases here.

Reply:
You are absolutely right. I was simply overly careful and did hope to sneak in an easter egg. Damn, you found it 😄